### ISSUE TITLE
Error UNKNOWN_MEMBER_ID and ILLEGAL_GENERATION

### ISSUE THREAD

**Issue by sebastian-alfers** [akka/reactive-kafka#152]
I upgraded from 0.10 to latest branch 0.11.M3 and get the following errors:

2016-05-23 16:13:54.791 [Writer-akka.kafka.default-dispatcher-18] ERROR o.a.k.c.c.i.ConsumerCoordinator - Error ILLEGAL_GENERATION occurred while committing offsets for group MyConsumerGroup

2016-05-23 16:15:00.656 [Writer-akka.kafka.default-dispatcher-77] ERROR o.a.k.c.c.i.ConsumerCoordinator - Error UNKNOWN_MEMBER_ID occurred while committing offsets for group MyConsumerGroup

Any idea? I did not change configs to the previous setup

Thanks

**Comment by jeroentervoorde** [comment]
We had the same errors and in our case it was caused by session timeouts in the consumer. The consumer buffers records which are then passed, on demand, up the stream. This makes it difficult to control the amount of processing time between polls (by default the timeout is 30 seconds) 

We fixed it by upgrading to kafka-clients 0.10 and limiting the number of records read by the poll() in the consumer using max.poll.records (only available in the 0.10 consumer).

Upgrading to kafka-clients 0.10 requires 2 small code changes so you will need to publish a local snapshot version.
